lua/telescope/builtin/files.lua

```diff
@@ -186,6 +186,26 @@ files.find_files = function(opts)
           table.insert(find_command, 2, v)
         end
       end
+    elseif 1 == vim.fn.executable("where") then
+      print('using the `where` function')
+      find_command = { 'where', '/r', '.', '*'}
+      if hidden then
+        print('The `hidden` key is not available for the Windows `where` command in `find_files`.')
+      end
+      if follow then
+        print('The `follow` key is not available for the Windows `where` command in `find_files`.')
+      end
+      if search_dirs then
+        table.remove(find_command, 3)
+        local dir_num = 1
+        for _,v in pairs(search_dirs) do
+          if dir_num == 1 then
+            table.insert(find_command, 3, v)
+          else
+            print('Multiple `search_dirs` are not available for the Windows `where` command in `find_files`.')
+          end
+        end
+      end
```

Comment:
hmm `search_dirs` are explicitly only meant for multiple directories. For single dir `cwd` should be preferred. Its set on `Job` level, so it should work without more work.

Reply:
> hmm `search_dirs` are explicitly only meant for multiple directories. For single dir `cwd` should be preferred. 

Okay, so would be best to remove this `search_dirs` bit and replace with a similar message to the other parts, right?

> Its set on `Job` level, so it should work without additional work.

I'm not really sure what you mean by this, could you elaborate?

Reply:
sure.

we do `finder = finders.new_oneshot_job(find_command, opts),` which will start a job (uv.spawn + more) https://github.com/nvim-telescope/telescope.nvim/blob/master/lua/telescope/finders/async_oneshot_finder.lua#L16

This is simplified. You end up with something like this:
```lua
local Job = require("plenary.job")

-- obv we dont have sync here
local result = Job:new({ 'where', '/r', '.', '*', cwd = cwd1}):sync()
```

But now you can change the `cwd1` var to change the directory and thats what happens with `Telescope find_files cwd=...` similar applies to other pickers. If you only wanna change one cwd, use cwd :) its just not documented, smith missed it when reading code, but thats fine

Feel free to ask more questions when you run into them :)

Ref job: https://github.com/nvim-lua/plenary.nvim/blob/master/lua/plenary/job.lua
Ref spec: https://github.com/nvim-lua/plenary.nvim/blob/master/tests/plenary/job_spec.lua

Reply:
Ah, I understand now, thanks 👍